baseball/js/app.js

```diff
@@ -0,0 +1,366 @@
+var width = 767;
+var height = 400;
+
+//Projection
+var projection = d3.geoAlbers()
+	.translate([0,0]);
+
+//Path
+var path = d3.geoPath(projection);
+
+
+var svg = d3.select('#map')
+	.append('svg')
+	.attr('width', width)
+	.attr('height', height);
+
+d3.select('#legend')
+	.append('p').text('Batting Averages by Stadium');
+
+var legendSVG = d3.select('#legend')
+	.append('svg')
+	.attr('width', 400)
+	.attr('height', 40);
+
+var legendSquareSize = 80;
+
+var zoom_map = d3.zoom()
+	.scaleExtent([1, 5.0])
+	.translateExtent([
+		[-400, -400], //top left corner
+		[400, 400] //bottom right corner
+	])
+	.on('zoom', function(){
+	
+		var offset = [
+			d3.event.transform.x, 
+			d3.event.transform.y
+		];
+
+		var scale = d3.event.transform.k * 850;
+
+		projection.translate(offset)
+			.scale(scale);
+
+		svg.selectAll('path')
+			.attr('d', path);
+
+		svg.selectAll('circle')
+			.attr('cx', function(d){
+				return projection([d.lon, d.lat])[0];
+			})
+			.attr('cy', function(d){
+				return projection([d.lon, d.lat])[1];
+			});
+});
+
+var map = svg.append('g')
+	.attr('id', 'map')
+	.call(zoom_map)
+	.call(
+		zoom_map.transform,
+		d3.zoomIdentity
+			.translate(width / 2, height /2)
+			.scale(1)
+	);
+
+map.append('rect')
+	.attr('x', 0)
+	.attr('y', 0)
+	.attr('width', width)
+	.attr('height', height)
+	.attr('opacity', 0);
+
+var color = d3.scaleLinear()
+	.domain([5, 10, 20, 30, 40])
+	.range(["#0454AE", "#15ABDD", "#5BA049", "#FF8833", "#D6202A"])
```

Comment:
I might suggest reversing the color scale (from a DS perspective). If I understand it correctly, high batting averages are better, but the highest number is red which I normally relate to something bad. I like that you scale the circles, too.

Reply:
Only argument is when some one is batting poorly they are considered cold at the plate and when they are killing it they are red hot.  Good to have a different perspective though.  Color choices always seem to be difficult.